src/main.py

```diff
@@ -97,71 +116,75 @@ def run(options: Options) -> int:
                         file=sys.stderr,
                     )
                     return 1
-                functions.append(list(all_functions.values())[index_or_name])
+                funcs.append(list(all_functions.values())[index_or_name])
             else:
                 if index_or_name not in all_functions:
                     print(f"Function {index_or_name} not found.", file=sys.stderr)
                     return 1
-                functions.append(all_functions[index_or_name])
+                funcs.append(all_functions[index_or_name])
 
     fmt = options.formatter()
-    function_names = set(all_functions.keys())
     typepool = TypePool(
         unknown_field_prefix="unk_" if fmt.coding_style.unknown_underscore else "unk",
         unk_inference=options.unk_inference,
     )
-    global_info = GlobalInfo(asm_data, arch, function_names, typemap, typepool)
+    global_info = GlobalInfo(
+        asm_data, arch, set(all_functions.keys()), typemap, typepool
+    )
+
+    functions: Dict[str, Function] = {f.name: f for f in funcs}
+    function_infos: Dict[str, Union[FunctionInfo, Exception]] = {}
+    flow_graphs: Dict[str, Union[FlowGraph, Exception]] = {}
 
-    flow_graphs: List[Union[FlowGraph, Exception]] = []
-    for function in functions:
+    function_names: List[str] = list(functions.keys())
+    function = get_next_func(function_names, functions, global_info)
+    while function:
         try:
-            flow_graphs.append(build_flowgraph(function, global_info.asm_data))
+            flow_graph = build_flowgraph(function, global_info.asm_data)
         except Exception as e:
             # Store the exception for later, to preserve the order in the output
-            flow_graphs.append(e)
+            flow_graph = e  # type: ignore
```

Comment:
We don't use `# type: ignore` in this codebase, `flow_graph` should have type `Union[FlowGraph, Exception]`